Cabal/Distribution/Simple.hs

```diff
@@ -263,6 +264,27 @@ buildAction hooks flags args = do
                (return lbi { withPrograms = progs })
                hooks flags' { buildArgs = args } args
 
+showBuildInfoAction :: UserHooks -> BuildFlags -> Args -> IO ()
+showBuildInfoAction hooks flags args = do
+  distPref <- findDistPrefOrDefault (buildDistPref flags)
+  let verbosity = fromFlag $ buildVerbosity flags
+      flags' = flags { buildDistPref = toFlag distPref }
+
+  lbi <- getBuildConfig hooks verbosity distPref
+  progs <- reconfigurePrograms verbosity
+             (buildProgramPaths flags')
+             (buildProgramArgs flags')
+             (withPrograms lbi)
+
+  pbi <- preBuild hooks args flags'
+  let lbi' = lbi { withPrograms = progs }
+      pkg_descr0 = localPkgDescr lbi'
+      pkg_descr = updatePackageDescription pbi pkg_descr0
+  -- TODO: Somehow don't ignore build hook?
```

Comment:
what about this TODO? What is the consequence of ignoring build hooks, i.e. what's the worst that can happen?

Reply:
Why are even build hooks being ignored?